### ISSUE TITLE
Active catalog/schema behaviour

### ISSUE THREAD

**Issue by harryshepard** [dbeaver/dbeaver#8059]
#### System information: 
- Operating system (distribution) and version
Windows 10
- DBeaver version
7

#### Connection specification:
- Database name and version
MariaDB

#### Describe the problem you're observing:
The active schema keeps randomly changing. I have auto sync connection with navigator on but it just keeps changing. I've added a key binding for "set as default", so when I press that, it is bolding the selected database and also updates the active at the top. Then randomly from time to time the same one is bold on the navigator but the active one has changed. I haven't figured out how to reproduce it yet but will update if I do. It should never be different from the one I set as default unless I specifically tell it to do so. I created a table in the wrong schema earlier because this is new and unexpected behaviour

**Comment by harryshepard** [comment]
This is happening after I run a query in the script editor. I change the active schema, run a query, then it's just changing to a different schema which I had default earlier. It's like it's remembering the first one I defaulted then reverting to that.

Even more strangely, I have just restarted and selected the one I actually want as default, then set a different one as default, ran a query in that schema. The active drop down at the top changed to the one I set first, but the queries are still running against the other one. It is very strange behaviour.

When I set as default in the navigator it should set any script files to the default unless they have been manually changed. I think the active schema should have a "use default" option

**Comment by serge-rider** [comment]
Since 6.3 each SQL editor has its own active schema. You can change it in the main toolbar (or by ctrl+0).
You can disable this behavior by turning SQL Editor option "Open separate connection for each editor" off.

**Comment by harryshepard** [comment]
Hi Serge

I want this feature to stay as it is. The problem is, it is changing when
I'm not telling it to. If I don't select one manually it should use the
default

**Comment by serge-rider** [comment]
@harryshepard I can't reproduce such "random" schema change.
Maybe it is a result of "auto sync connection" option on? It changes editor's connection whenever you change selection in database navigator.

**Comment by harryshepard** [comment]
@serge-rider I turned that on to try and stop this odd behaviour. It may have something to do with my user shortcut of F3 to set a schema as the default (which is changing the editor schema as expected).

What other settings could change the editor schema? In general I will just always select the schema I want to run queries against and hit F3 so I would like for the editor to just always use the default as I set it. I am clicking around between the navigator and editor text area, and I've managed to get the editor schema to be different from the default schema (bold in the navigator) again somehow. When I press F3 on another schema it sets it correctly again but I definitely still have a problem of the editor's active schema changing from the default I'm setting within the navigator
